src/ILCompiler.Compiler/src/CppCodeGen/CppWriter.cs

```diff
@@ -93,6 +90,21 @@ private void BuildExternCSignatureMap()
             }
         }
 
+        private IEnumerable<string> GetParameterNamesForMethod(MethodDesc method)
+        {
+            // TODO: The uses of this method need revision. The right way to get to this info is from
+            //       a MethodIL. For declarations, we don't need names.
```

Comment:
This part makes me particularly unhappy, but it wasn't immediately clear why the code path that ends up needing parameter names is called from half a dozen places within CppCodegen. I'm not sure this will play well with the various strategies where a body of a MethodDesc is replaced with a body of a different MethodDesc, unless we want to enforce the parameter names match so that CppCodegen won't notice.